### PR TITLE
Update `CHANGELOG.md`

### PR DESCRIPTION
**Pull request by Urhengulas**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,19 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 ## [Unreleased]
 
 - [#]
+- [#621]: Readme Diagram: Replace duplicate defmt-itm with defmt-rtt
+- [#617]: Add display hint to output `u64` as ISO8601 time
+- [#620]: Tidy up
```
Comment (japaric) [inline CHANGELOG.md:13]: nit: I would expand the description of this PR a bit more so one can understand better without reading the PR description. maybe 'remove unused code / dependencies'

**Review by japaric — APPROVED** [review]
reads good to me module one nit

File: CHANGELOG.md (hunk 1 above)
Comment (Urhengulas) [reply CHANGELOG.md:13]:
```suggestion
- [#620]: Tidy up
```
```suggestion
- [#620]: Tidy up: remove unused code and dependencies
```

Comment (Urhengulas) [reply CHANGELOG.md:13]:
```suggestion
- [#620]: Tidy up: remove unused code and dependencies
```

**Comment by Urhengulas** [comment]
bors r=japaric

**Pull request merged**